### PR TITLE
COL-238 User Registration API for Mobile

### PR DESCRIPTION
**Pull request by imhassantariq**
Register user from AuthExchangeView if the user already doesn't exist. 

Ticket Link: https://edlyio.atlassian.net/browse/COL-238

### CODE REVIEW

File: openedx/features/colaraz_features/auth_backend.py [hunk 1]
```diff
@@ -69,3 +71,22 @@ def get_user_details(self, response):
...
+        # Imported here to avoid circular imports
+        from third_party_auth.utils import user_exists
+
+        user_data = super(ColarazIdentityServer, self).user_data(access_token, *args, **kwargs)
+        request = get_current_request()
+        request_source = request.GET.get('src')
+
+        if (request.view_name == "AccessTokenExchangeView" and request_source == 'mobile'
+                and not user_exists(user_data)):
+            LOGGER.info("Received request from mobile registering a new user.")
```
Comment (mumerfarooq000) [inline openedx/features/colaraz_features/auth_backend.py:89]:
```suggestion
            LOGGER.info("Received request from mobile and now registering a new user.")
```

File: openedx/features/colaraz_features/auth_backend.py [hunk 2]
```diff
@@ -69,3 +71,22 @@ def get_user_details(self, response):
...
+        """
+
+        # Imported here to avoid circular imports
+        from third_party_auth.utils import user_exists
+
+        user_data = super(ColarazIdentityServer, self).user_data(access_token, *args, **kwargs)
+        request = get_current_request()
+        request_source = request.GET.get('src')
+
+        if (request.view_name == "AccessTokenExchangeView" and request_source == 'mobile'
```
Comment (mumerfarooq000) [inline openedx/features/colaraz_features/auth_backend.py:87]: its better according to [pep8](https://stackoverflow.com/a/53173/5579792) to end with a logical operator

**Review by mumerfarooq000 — APPROVED** [review]

File: openedx/features/colaraz_features/auth_backend.py (hunk 2 above)
Comment (imhassantariq) [reply openedx/features/colaraz_features/auth_backend.py]: Done

File: openedx/features/colaraz_features/auth_backend.py (hunk 1 above)
Comment (imhassantariq) [reply openedx/features/colaraz_features/auth_backend.py]: Done

**Pull request merged**